### PR TITLE
[react-dates@21.8.0] Add types for deep imports

### PR DESCRIPTION
**Pull request by mesaugat**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

## Changes

- Allows for deep imports from `lib` and `esm` path which is the suggested way to import components and utils that `react-dates` provides. Without deep imports, tree shaking is not possible and needs some tweaking in the bundler configuration.

**Note: No need to add tests as this is a simple import/re-export from `index.d.ts`**

### Current Scenario

```
import toISODateString from 'react-dates/esm/utils/toISODateString';
```

### Error

```
Could not find a declaration file for module 'react-dates/esm/utils/toISODateString'. `../node_modules/react-dates/esm/utils/toISODateString.js' implicitly has an 'any' type.
  If the 'react-dates' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-dates`
```

### CODE REVIEW

File: types/react-dates/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# @types/react-dates
+
+## Adding a new entry
+
+When a new component or util has been published by `react-dates`:
+
+- Add the type to `index.d.ts`
+- Add your entry import/re-export in `esm` & `lib` folders
+- Add `esm/<type>.d.ts` & `src/<type>.d.ts` to `OTHER_FILES.txt`
```
Comment (cgriebel) [inline types/react-dates/README.md:9]: this should this say `lib/<type>.d.ts`, not `src/<type>.d.ts` right?

Comment (cgriebel) [reply types/react-dates/README.md:9]: Other than that, looks good.

Comment (cgriebel) [inline types/react-dates/README.md:9]: This should this say lib/<type>.d.ts, not src/<type>.d.ts right?

**Review by cgriebel — CHANGES_REQUESTED** [review]

File: types/react-dates/README.md (hunk 1 above)
Comment (cgriebel) [reply types/react-dates/README.md:9]: Aside from this, looks good. Thanks for contributing!

Comment (mesaugat) [reply types/react-dates/README.md]: Yes. I've changed it to `lib/<type>.d.ts`

**Review by cgriebel — APPROVED** [review]

**Review by rbuckton — APPROVED** [review]

**Comment by cgriebel** [comment]
Ready to merge

**Pull request merged**